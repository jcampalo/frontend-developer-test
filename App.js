import React, { PureComponent } from 'react';
import styled from 'styled-components/native';
import {
 Platform, StatusBar
} from 'react-native';
import {
 AppLoading, Asset, Font, Icon
} from 'expo';
import AppNavigator from './src/navigation/AppNavigator';

const Wrapper = styled.View`
  flex: 1;
`;

class App extends PureComponent {
  state = {
    isLoadingComplete: false,
  };

  loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require('./assets/images/robot-dev.png'),
        require('./assets/images/robot-prod.png'),
      ]),
      Font.loadAsync({
        // This is the font that we are using for our tab bar
        ...Icon.Ionicons.font,
        // We include SpaceMono because we use it in HomeScreen.js. Feel free
        // to remove this if you are not using it in your app
        'space-mono': require('./assets/fonts/SpaceMono-Regular.ttf'),
      }),
    ]);
  }

  handleLoadingError = error => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    console.warn(error);
  }

  handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  }

  render() {
    if (!this.state.isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this.loadResourcesAsync}
          onError={this.handleLoadingError}
          onFinish={this.handleFinishLoading}
        />
      );
    }

    return (
      <Wrapper>
        {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
        <AppNavigator />
      </Wrapper>
    );
  }
}

export default App;
