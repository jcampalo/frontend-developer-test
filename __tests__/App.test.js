import 'react-native';
import React from 'react';
import { create } from 'react-test-renderer';

import App from '../App';

jest.mock('../src/RootComponent', () => 'RootComponent');

describe('App', () => {
  describe('render', () => {
    it('should render loading', () => {
      const tree = create(<App />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render root component when skip', () => {
      const tree = create(
        <App skipLoadingScreen />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
