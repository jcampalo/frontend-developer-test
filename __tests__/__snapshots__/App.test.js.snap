// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render should render loading 1`] = `null`;

exports[`App render should render root component when skip 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RootComponent />
</View>
`;
